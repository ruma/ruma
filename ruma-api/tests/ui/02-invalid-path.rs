use ruma_api::ruma_api;

ruma_api! {
    metadata: {
        description: "Does something.",
        method: GET,
        name: "newtype_body_endpoint",
        path: "µ / ° / § / €",
        rate_limited: false,
        requires_authentication: false,
    }

    request: {
        #[ruma_api(query_map)]
        pub fields: Vec<(String, String)>,
    }

    response: { }
}

fn main() {}
