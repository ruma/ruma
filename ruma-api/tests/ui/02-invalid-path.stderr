error: path may only contain printable ASCII characters with no spaces
 --> $DIR/02-invalid-path.rs:8:15
  |
8 |         path: "µ/°/§/€",
  |               ^^^^^^^^^

error: path may only contain printable ASCII characters with no spaces
  --> $DIR/02-invalid-path.rs:26:15
   |
26 |         path: "path/to/invalid space/endpoint",
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
